src/test/java/de/tum/in/www1/artemis/ExamIntegrationTest.java

```diff
@@ -1696,9 +1710,74 @@ private int getNumberOfProgrammingExercises(Exam exam) {
         return count;
     }
 
-    @Test
+    private void configureCourseAsBonusWithIndividualAndTeamResults(Course course, GradingScale bonusToGradingScale) {
+        ZonedDateTime pastTimestamp = ZonedDateTime.now().minusDays(5);
+        Long courseId = course.getId();
+        TextExercise textExercise = database.createIndividualTextExercise(course, pastTimestamp, pastTimestamp, pastTimestamp);
+        Long individualTextExerciseId = textExercise.getId();
+        TextExercise exerciseWithoutParticipants = database.createIndividualTextExercise(course, pastTimestamp, pastTimestamp, pastTimestamp);
+
+        Exercise teamExercise = database.createTeamTextExercise(course, pastTimestamp, pastTimestamp, pastTimestamp);
+        User student1 = userRepo.findOneByLogin("student1").get();
+        User tutor1 = userRepo.findOneByLogin("tutor1").get();
+        Long teamTextExerciseId = teamExercise.getId();
+        Long team1Id = database.createTeam(Set.of(student1), tutor1, teamExercise, "team1").getId();
+        User student2 = userRepo.findOneByLogin("student2").get();
+        User student3 = userRepo.findOneByLogin("student3").get();
+        User tutor2 = userRepo.findOneByLogin("tutor2").get();
+        Long team2Id = database.createTeam(Set.of(student2, student3), tutor2, teamExercise, "team2").getId();
+
+        // Creating result for student1
```

Comment:
Those comments are redundant as the function names are _very_ verbose.

Reply:
I removed these redundant comments and replaced one with a more explanatory one here: https://github.com/ls1intum/Artemis/pull/5668/commits/b712620b92d980bf9de5f35c0f6d2cb8e080847a